### PR TITLE
fix: adapt gemm collector for zig-zag pattern

### PR DESCRIPTION
**Pull request by Harrilee**
### Overview

Change gemm collection base on observed zig-zag pattern.
image

image

### TODO
Collect data on latest TRTLLM version
- [ ] L40S
- [ ] A100
- [x] H100
- [ ] H200
- [ ] GB200 (1.2.0rc5)
- [ ] GB200 (1.2.0rc6)
- [ ] GB300 (1.2.0rc5)
- [ ] GB300 (1.2.0rc6post3)
- [ ] B200

## Summary by CodeRabbit

* **Tests**
  * Improved test case generation to provide more comprehensive coverage of edge cases and patterns.

### CODE REVIEW

File: collector/common_test_cases.py [hunk 1]
```diff
@@ -147,29 +147,16 @@ class GemmCommonTestCase:
...
-        256,
-        384,
-        512,
-        768,
-        1024,
-        2048,
-        4096,
-        8192,
-    ]
+    x_list = [1, 32, 64, 128]
```
Comment (YijiaZhao) [inline collector/common_test_cases.py]: Small x will not cost too much time, we need dense sampling
for example:
if x < 32:
x = [i for i in range(32)]
elif x < 128
x += [i for i in range(32, 128, 16)] + [i+1 for i in range(32, 128, 16)]
elif x < 256
x += [i for i in range(32, 256, 32)] + [i+1 for i in range(32, 256, 32)]

Comment (Harrilee) [reply collector/common_test_cases.py]: How about we keep it as is?
A.k.a `x_list = [1, 2, 4, 8, 16, 32, 48, 64, 80, 96, 128, 160, 192]`

Comment (YijiaZhao) [reply collector/common_test_cases.py]: small x also has zigzag prroblem, I suggest:
if x < 16:
x = [i for i in range(16)]
elif x < 128
x += [i for i in range(16, 128, 16)] + [i+1 for i in range(16, 128, 16)]
elif x < 256
x += [i for i in range(32, 256, 32)] + [i+1 for i in range(32, 256, 32)]

but you need to check the boundary conditions

**Review by tianhaox — APPROVED** [review]

**Review by jasonqinzhou — APPROVED** [review]
